### PR TITLE
Add DataSourceMetadataOperator and DataSourceManager

### PR DESCRIPTION
**Pull request by shreyabiradar07**
## Description

This PR has the following changes: 

- Adds `DataSourceMetadataOperator` (yet to implement CRUD operations)
- Implementation of `DataSourceManager` on top of metadata operator

### Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

**Test Configuration**
* Kubernetes clusters tested on: minikube

## Checklist :dart:

- [x] Followed coding guidelines
- [x] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

## Additional information

This PR has a dependency on [#1139](https://github.com/kruize/autotune/pull/1139) to be merged first, as it uses `DataSourceInfo`
Implementation of CRUD operations for `DataSourceMetadataOperator` will be in next PR

### CODE REVIEW

**Comment by dinogun** [comment]
@shreyabiradar07 These needs to be rebased with the latest mvp_demo branch

File: src/main/java/com/autotune/common/datasource/DataSourceMetadataOperator.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    TODO - Implement methods to support update and delete operations for periodic update of DataSourceMetadataInfo
+    public DataSourceMetadataInfo updateDataSourceMetadata(DataSourceInfo dataSource) {
+
+    }
+
+    public void deleteDataSourceMetadata(DataSourceInfo dataSource) {
+
+    }
+
+     */
```
Comment (dinogun) [inline src/main/java/com/autotune/common/datasource/DataSourceMetadataOperator.java:37]: As a rule, we do not merge code that is commented, please remove these lines

File: src/main/java/com/autotune/common/datasource/DataSourceManager.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            if (null != dataSource) {
+                throw new DataSourceNotExist(KruizeConstants.DataSourceConstants.DataSourceErrorMsgs.MISSING_DATASOURCE_INFO);
+            }
+            dataSourceMetadataOperator.deleteDataSourceMetadata(dataSource);
+        } catch (DataSourceNotExist e) {
+            LOGGER.error(e.getMessage());
+        }
+    }
+
+     */
```
Comment (dinogun) [inline src/main/java/com/autotune/common/datasource/DataSourceManager.java:83]: Remove commented code

Comment (shreyabiradar07) [reply src/main/java/com/autotune/common/datasource/DataSourceManager.java]: Done, PR is ready to be merged

**Review by dinogun — APPROVED** [review]
LGTM - CRUD PR should have all of the functionality

**Pull request merged**